pkg/servicemanager/startuprecorder/startuprecorder.go

```diff
@@ -0,0 +1,71 @@
+package startuprecorder
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type ServiceData struct {
+	Name         string        `json:"name"`
+	Dependencies []string      `json:"dependencies"`
+	Start        time.Time     `json:"start"`
+	Ready        time.Time     `json:"ready"`
+	TimeToReady  time.Duration `json:"timeToReady"`
+}
+
+type StartupRecorder struct {
+	Services              []ServiceData `json:"services"`
+	MicroshiftStart       time.Time     `json:"microshiftStart"`
+	MicroshiftReady       time.Time     `json:"microshiftReady"`
+	MicroshiftTimeToReady time.Duration `json:"microshiftTimeToReady"`
```

Comment:
You could drop the `Microshift` from these if you'd create another struct, but I'll leave it up to you.
```
{
    "services": [
        {...}
    ]
    "microshift": {
        "start": ...
        "ready": ...,
        "timeToReady": ...
    }
}

```